Lib/test/test_functools.py

```diff
@@ -2305,6 +2305,13 @@ def _(arg: typing.Iterable[str]):
         ))
         self.assertTrue(str(exc.exception).endswith(msg_suffix))
 
+    def test_invalid_positional_argument(self):
+        @functools.singledispatch
+        def f(*args):
+            pass
+        msg = 'f requires at least 1 positional argument'
+        with self.assertRaisesRegexp(TypeError, msg):
```

Comment:
`assertRaisesRegexp()` is deprecated, so this test fails when test_functools is run with `-Werror` (see #8261).